### PR TITLE
Allow underscores in routes

### PR DESCRIPTION
**Pull request by ElijahAhianyo**
### Description
This pull request addresses the need for converting underscores to hyphens in routes, which was previously implemented. The goal of this change is to enhance the flexibility of route naming by allowing both underscores and hyphens to coexist without the need for any conversions. This PR proposes to remove the underscore-to-hyphen conversion logic, allowing developers to freely choose between these two naming conventions for routes.

### Changes Made
* Removed the underscore-to-hyphen conversion logic from route handling, enabling route names to support both underscores and hyphens directly.

### Motivation
* Allowing both underscore-separated and hyphen-separated route names provides developers with greater flexibility in choosing naming conventions that align with their preferences and practices.
* By removing the conversion, we empower developers to use the naming convention that best suits their needs, contributing to a more adaptable and user-friendly codebase.

### Impact
With this change, developers are free to use both underscores and hyphens as they see fit in route names.
The removal of the underscore-to-hyphen conversion ensures that developers can choose their preferred naming convention without the need for special handling.
Existing routes with either underscores or hyphens in their names will continue to function as expected.

### CODE REVIEW

**Review by masenf — DISMISSED** [review]
Any plans to also make upper case letters work? That seems to come up about every 2-3 weeks on discord.

**Comment by ElijahAhianyo** [comment]
> Any plans to also make upper case letters work? That seems to come up about every 2-3 weeks on discord.

Yeah, could add that to this PR

File: reflex/utils/format.py [hunk 1]
```diff
@@ -210,9 +210,10 @@ def format_route(route: str) -> str:
     Returns:
         The formatted route.
     """
-    # Strip the route.
     route = route.strip("/")
-    route = to_snake_case(route).replace("_", "-")
+    # Strip the route and format casing.
+    if format_case:
```
Comment (picklelo) [inline reflex/utils/format.py:215]: Let's make a function called `to_kebab_case` and call it here

File: integration/test_event_chain.py [hunk 2]
```diff
@@ -97,11 +97,11 @@ def click_yield_nested(self):
 
         def redirect_return_chain(self):
             self.event_order.append("redirect_return_chain")
-            yield rx.redirect("/on-load-return-chain")
+            yield rx.redirect("/on_load_return_chain")
```
Comment (picklelo) [inline integration/test_event_chain.py:100]: This file shouldn't need to change

**Review by picklelo — CHANGES_REQUESTED** [review]
Looks good! I think we just need to update the integration tests in integration/test_dynamic_routes.py

File: integration/test_dynamic_routes.py [hunk 3]
```diff
@@ -69,7 +69,7 @@ def redirect_page():
 
 @pytest.fixture(scope="session")
 def dynamic_route(
-    app_harness_env: Type[AppHarness], tmp_path_factory
+        app_harness_env: Type[AppHarness], tmp_path_factory
```
Comment (masenf) [inline integration/test_dynamic_routes.py:72]: i don't understand why these blocks are all indented more...

Comment (ElijahAhianyo) [reply integration/test_dynamic_routes.py:72]: probably as a result of running precommits (black), not typically sure why those parts are affected

**Review by picklelo — APPROVED** [review]
looks good!

**Pull request merged**